_posts/2016-01-15-GitHubAnimator.md

```diff
@@ -0,0 +1,62 @@
+
+#### CREATING THE PRESENTATION
+
+Initially I used Office automation frameworks by Microsoft to generate a PowerPoint presentation, but decided that using a specific application could be too restrictive. So I started looking for a presentation framework using HTML5 and came across  [Reveal.js](http://lab.hakim.se/reveal-js/#/) and it looked like a perfect match for what I wanted to build..
+
+##### Hightlighting the Code Samples
+After playing around with Reveal.js I found out that the code highlight features, out of the box, were a bit limited for what I wanted. So I searched for a framework that would work with Reveal.js. I found [Prism.js](http://prismjs.com/) did all I wanted and a bit more. I set out to integrate Prism.js with Reveal.js and finally had something that would meet my requirements.
+
+##### Fiddling With CSS
+Last part was to make some changes to Reveal.js and Prism.js CSS's to get the desired visual effect. This is an area I'm a bit out of my comfort zone so it took me a while to get something half decent.
+
+### The End Result
+The code, as it stands now, is still very crude but is capable of retrieving all commits, for a file, in a given repository, and generate a presentation using Reveal.js. The presentation has a slide for each commit that changed the specified file. The slides use Prism.js to highlight the code.
+
+There are still a few things hardcoded on the F# code, so it's not a "consumer" ready product, but any developer should be able to use it, as is now.
+
+### How Can I Use It?
+
+Start by grabbing the code to GitHubAnimator from [here](https://github.com/pedromsantos/GitHubAnimator)
+
+You should be using a GitHub [Personal access token](https://github.com/settings/tokens). You can use the GitHubAPI as an anonymous user but the request limits are much lower. Make sure you copy the token. Then create an environment variable called githubtoken and set the GitHubToken you got from GitHub as a value. Be careful with the personal access token since it is effectively a replacement for your username and password.
+
+For the time being you need to make a few changes to the code if you want to use GitHubAnimator. I started building a console interface but its still in its infancy. So go to the integration tests module and change this line:
+
+```FSharp
+let parameters = { Owner = "pedromsantos"; Repository = "FSharpKatas"; File="Bowling.fs"; Language = "language-fsharp" }
+```
+
+Note: Language should be one the the languages prism.js supports and should be in the format language-xxx [Prism languages](http://prismjs.com/#languages-list)
+
+Then go to the last integration test change the hardcoded paths, unignore it and finally execute that test manually. This should generate a Reveal.js presentation in the folder you specified.
+
+The code, as is, is good enough for me :) But if you want to make it friendlier feel free to make a pull request. I make no promises on when, if ever, I'm changing it :)
```

Comment:
I feel that this howto belongs in the readme of the repository more than in this blogpost....